db/src/schema.rs

```diff
@@ -49,6 +49,9 @@ fn validate_schema_map(entid_map: &EntidMap, schema_map: &SchemaMap) -> Result<(
         if attribute.component && attribute.value_type != ValueType::Ref {
             bail!(ErrorKind::BadSchemaAssertion(format!(":db/isComponent true without :db/valueType :db.type/ref for entid: {}", ident())))
         }
+        if attribute.cached && !attribute.index {
+            bail!(ErrorKind::BadSchemaAssertion(format!(":db/cached true without :db/index true for entid: {}", ident())))
```

Comment:
Could you explain this relation?

AIUI, `index` means that value lookups are fast. All datoms are indexed by attribute; `index: true` means they're also indexed by value. I think caching goes the other way, no — known `a`, build a map from `e` to `v`?

Reply:
I was thinking that if, during cache creation/invalidation I would need to quickly look up the attribute. I guess I overlooked that it was the value that was being indexed and not the attribute itself. I'll revert that.